target_postgres/postgres.py

```diff
@@ -391,24 +416,42 @@ def get_postgres_datetime(self, *args):
         return parsed_datetime.format('YYYY-MM-DD HH:mm:ss.SSSSZZ')
 
     def add_column(self, cur, table_name, column_name, column_schema):
+        canonicalized_name = self.canonicalize_identifier(column_name)
+
+        if canonicalized_name != column_name:
+            self.add_column_mapping(cur, table_name, column_name, canonicalized_name, column_schema)
+
         data_type = json_schema.to_sql(column_schema)
 
         if not json_schema.is_nullable(column_schema) \
                 and not self.is_table_empty(cur, table_name):
             self.logger.warning('Forcing new column `{}.{}.{}` to be nullable due to table not empty.'.format(
                 self.postgres_schema,
                 table_name,
-                column_name))
+                canonicalized_name))
             data_type = json_schema.to_sql(json_schema.make_nullable(column_schema))
 
         to_execute = sql.SQL('ALTER TABLE {table_schema}.{table_name} ' +
                              'ADD COLUMN {column_name} {data_type};').format(
             table_schema=sql.Identifier(self.postgres_schema),
             table_name=sql.Identifier(table_name),
-            column_name=sql.Identifier(column_name),
+            column_name=sql.Identifier(canonicalized_name),
             data_type=sql.SQL(data_type))
 
-        cur.execute(to_execute)
+        try:
+            cur.execute(to_execute)
+        except Exception as ex:
+            ## Name collision with non mapped, non canonicalized name
+            if column_name != canonicalized_name:
```

Comment:
Is there a chance that an exception could be thrown and the statement to be true for reasons other than a name collision? Is there anything in the exception or error message we could inspect?

Reply:
So, in this _very specific_ instance we know:
- an error has been thrown trying to execute a column ***add*** to remote
- the `column_name` we have present in our _streamed_ schema, ***does not*** match the `canonicalized_name` we've generated

In this case, we can safely assume the the likelihood of some collision having happened to be pretty great (ie, I think we can add the context to the end user and it'd be beneficial).

To that point though, I think this can be made _more generic_ and still be beneficial. I'll throw in an exception chain here and modify the message to remove the `if` and we can get the same benefit to the end user.